wolfcrypt/test/test.c

```diff
@@ -7255,6 +7265,151 @@ int ecc_encrypt_test(void)
+
+    /* extract the client key from der buffer in wolfssl/certs_test.h */
+    XMEMCPY(tmp, ecc_clikey_der_2048, sizeof_ecc_clikey_der_2048);
+    bytes = sizeof_ecc_clikey_der_2048;
+
+    /* place client key into ecc_key struct cliKey */
+    ret = wc_EccPrivateKeyDecode(tmp, &idx, &cliKey, (word32)bytes);
+    if (ret != 0) {
+        TEST_XFREE(tmp, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+        return -41;
+    }
+
+    XMEMSET(tmp, 0, FOURK_BUF);
+    idx = 0;
+
+    /* extract the server key from der buffer in wolfssl/certs_test.h */
+    XMEMCPY(tmp, ecc_key_der_2048, sizeof_ecc_key_der_2048);
+    bytes = sizeof_ecc_key_der_2048;
+
+    /* place server key into ecc_key struct servKey */
+    ret = wc_EccPrivateKeyDecode(tmp, &idx, &servKey, (word32)bytes);
+    if (ret != 0) {
+        TEST_XFREE(tmp, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+        return -41;
+    }
+
+    ret = wc_InitRng(&rng);
+    if (ret != 0) {
+        TEST_XFREE(tmp, HEAP_HINT ,DYNAMIC_TYPE_TMP_BUFFER);
+        return -42;
+    }
+
+#if defined(HAVE_ECC_ENCRYPT) && defined(HAVE_HKDF)
+    {
+        word32 y;
+        /* test encrypt and decrypt if they're available */
+        x = sizeof(out);
+        ret = wc_ecc_encrypt(&cliKey, &servKey, in, sizeof(in), out, &x, NULL);
+        if (ret < 0) {
+            TEST_XFREE(tmp, HEAP_HINT ,DYNAMIC_TYPE_TMP_BUFFER);
+            return -43;
+        }
+
+        y = sizeof(plain);
+        ret = wc_ecc_decrypt(&cliKey, &servKey, out, x, plain, &y, NULL);
+        if (ret < 0) {
+            printf("ret = %d\n", ret);
+            TEST_XFREE(tmp, HEAP_HINT ,DYNAMIC_TYPE_TMP_BUFFER);
+            return -44;
+        }
+
+        if (XMEMCMP(plain, in, inLen)) {
+            TEST_XFREE(tmp, HEAP_HINT ,DYNAMIC_TYPE_TMP_BUFFER);
+            return -45;
+        }
+    }
+#endif
+
+
+    x = sizeof(out);
+    ret = wc_ecc_sign_hash(in, inLen, out, &x, &rng, &cliKey);
+    if (ret < 0) {
+        TEST_XFREE(tmp, HEAP_HINT ,DYNAMIC_TYPE_TMP_BUFFER);
+        return -46;
+    }
+    XMEMSET(plain, 0, sizeof(plain));
+
+    ret = wc_ecc_verify_hash(out, x, plain, sizeof(plain), &verify, &cliKey);
+    if (ret < 0) {
+        TEST_XFREE(tmp, HEAP_HINT ,DYNAMIC_TYPE_TMP_BUFFER);
+        return -47;
+    }
+
+    if (XMEMCMP(plain, in, ret)) {
+        TEST_XFREE(tmp, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+        return -48;
+    }
+
+    XMEMCPY(tmp, cliecc_cert_der_2048, sizeof_cliecc_cert_der_2048);
+    bytes = sizeof_cliecc_cert_der_2048;
+
+#ifdef WOLFSSL_TEST_CERT
+    InitDecodedCert(&cert, tmp, (word32)bytes, 0);
+
+    ret = ParseCert(&cert, CERT_TYPE, NO_VERIFY, 0);
+    if (ret != 0) return -491;
+
+    FreeDecodedCert(&cert);
+#else
+    (void)bytes;
+#endif
+
+#ifdef WOLFSSL_CERT_EXT
+    XMEMCPY(tmp, ecc_clikeypub_der_2048, sizeof_ecc_clikeypub_der_2048);
+    bytes = sizeof_clikeypub_der_2048;
+
+//----------------------------------------------------------------
+//    ret = wc_InitRsaKey(&keypub, 0);
```

Comment:
This and line 7401 below that are commented out. Is the WOLFSSL_CERT_EXT testing for ECC here yet to be implemented?